File: src/main/java/com/checkers/utils/BoardUtils.java
package com.checkers.utils;

import com.checkers.exceptions.ConvertCellException;

public final class BoardUtils {

    public static boolean isCoordinatesExists(int row, int column) {
        return isCoordinateExists(row) && isCoordinateExists(column);
    }

    public static boolean isCoordinateExists(int coordinate) {
        return coordinate >= 1 && coordinate <= 8;
    }

    public static int convertColumn(char columnName) {
        if (columnName < 'a' || columnName > 'h') {
            throw new ConvertCellException("Column '%s' doesn't exists".formatted(columnName));
        }

        return columnName - 'a' + 1;
    }

    public static int convertRow(char rowName) {
        if (rowName < '1' || rowName > '8') {
            throw new ConvertCellException("Row '%s' doesn't exists".formatted(rowName));
        }

        return rowName - '1' + 1;
    }
}

File: domain/src/main/kotlin/ru/kiseru/checkers/domain/utils/BoardUtils.kt
package ru.kiseru.checkers.domain.utils

import ru.kiseru.checkers.domain.exception.CellNotFoundException

fun getCellCaption(cell: Pair<Int, Int>): String =
    getCellCaption(cell.first, cell.second)

fun getCellCaption(row: Int, column: Int): String =
    if (isCoordinatesExists(row, column)) {
        val letter = ('a' + column - 1)
        letter.toString() + row
    } else {
        throw CellNotFoundException(row, column)
    }

fun isCoordinatesExists(cell: Pair<Int, Int>): Boolean =
    isCoordinateExists(cell.first) && isCoordinateExists(cell.second)

fun isCoordinatesExists(row: Int, column: Int): Boolean =
    isCoordinateExists(row) && isCoordinateExists(column)

fun isCoordinateExists(coordinate: Int): Boolean =
    coordinate in 1..8
